AmplifyPlugins/Auth/AWSAuthPlugin/Support/Utils/AuthErrorHelper.swift

```diff
@@ -0,0 +1,174 @@
+                                            AWSCognitoAuthError.softwareTokenMFANotEnabled)
+
+        case .tooManyFailedAttempts(let message):
+            return AmplifyAuthError.service(message,
+                                            AuthPluginErrorConstants.tooManyFailedError,
+                                            AWSCognitoAuthError.failedAttemptsLimitExceeded)
+
+        case .tooManyRequests(let message),
+             .limitExceeded(let message):
+            return AmplifyAuthError.service(message,
+                                            AuthPluginErrorConstants.tooManyRequestError,
+                                            AWSCognitoAuthError.requestLimitExceeded)
+
+        case .errorLoadingPage(let message):
+            return AmplifyAuthError.service(message,
+                                            AuthPluginErrorConstants.errorLoadingPageError,
+                                            AWSCognitoAuthError.errorLoadingUI)
+
+        case .deviceNotRemembered(let message):
+            return AmplifyAuthError.service(message,
+                                            AuthPluginErrorConstants.deviceNotRememberedError,
+                                            AWSCognitoAuthError.deviceNotTracked)
+
+        case .invalidState(let message):
+            return AmplifyAuthError.invalidState(message, AuthPluginErrorConstants.invalidStateError)
+
+        case .invalidConfiguration(let message),
+             .cognitoIdentityPoolNotConfigured(let message),
+             .invalidUserPoolConfiguration(let message):
+            return AmplifyAuthError.configuration(message, AuthPluginErrorConstants.configurationError)
+
+        case .notAuthorized(let message):
+            // Not authorized is thrown from server at different scenarios
+            return AmplifyAuthError.notAuthorized(message, AuthPluginErrorConstants.notAuthorizedError)
+
+        // Below error should not happen, these will be handled inside the plugin.
+        case .notSignedIn(let message), // Called in getTokens/getPassword when not signedin to CUP
+        .identityIdUnavailable(let message), // From getIdentityId. Handled in plugin
+        .guestAccessNotAllowed(let message), // Returned from getAWSCredentials. Handled in plugin
+        .federationProviderExists(let message), // User is already signed in to user pool in federatedSignIn
+        .unableToSignIn(let message), // Called in signout, releaseSignInWait.
+        .userCancelledSignIn(let message),
+        .idTokenNotIssued(let message), // Not used anywhere.
+        .badRequest(let message), // HostedUI returned with bad url or request.
+        .userPoolNotConfigured(let message):
+            return AmplifyAuthError.unknown(message)
+
+        case .securityFailed(let message):
+            // Error occured inside HostedUI
+            // TODO: Handle this error in Auth.signInWithUI
```

Comment:
Here & throughout: If this TODO isn't coming in an imminent PR, add a backlog issue & link it

Reply:
Added tracking number